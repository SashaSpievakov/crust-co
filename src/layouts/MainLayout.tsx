import { Outlet, useLocation } from 'react-router-dom';
import { useEffect, useRef } from 'react';

import { Wrapper, Main, Container } from '../styles/Base.styled';
import Footer from '../components/Footer/Footer';
import Header from '../components/Header/Header';

const MainLayout = () => {
  const scrollRef = useRef<null | HTMLDivElement>(null);
  const isMounted = useRef<boolean>(false);
  const location = useLocation();

  useEffect(() => {
    if (isMounted.current) {
      scrollRef.current?.scrollIntoView();
    }

    isMounted.current = true;
  }, [location.pathname]);

  return (
    <Wrapper ref={scrollRef}>
      <Header />
      <Main>
        <Container>
          <Outlet />
        </Container>
      </Main>
      <Footer />
    </Wrapper>
  );
};
export default MainLayout;
