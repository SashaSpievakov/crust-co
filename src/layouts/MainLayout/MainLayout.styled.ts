import styled from 'styled-components';

const Main = styled.main`
  padding: 40px 0;
  border-top: 1px solid ${({ theme }) => theme.colors.gray};
  border-bottom: 1px solid ${({ theme }) => theme.colors.gray};

  @media screen and (max-width: 550px) {
    padding: 30px 0;
  }
`;

export { Main };
