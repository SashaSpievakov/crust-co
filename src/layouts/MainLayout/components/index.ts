export { Footer } from './Footer';
export { Header } from './Header';
export { LayoutErrorBoundary } from './LayoutErrorBoundary';
