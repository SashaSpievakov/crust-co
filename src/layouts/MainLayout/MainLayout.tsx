import { FC } from 'react';
import { Outlet } from 'react-router-dom';

import { LayoutErrorBoundary } from '../LayoutErrorBoundary';
import { Footer } from './Footer';
import { Header } from './Header';
import { Main } from './MainLayout.styled';

export const MainLayout: FC = () => {
  return (
    <>
      <Header />
      <Main>
        <LayoutErrorBoundary>
          <Outlet />
        </LayoutErrorBoundary>
      </Main>
      <Footer />
    </>
  );
};
