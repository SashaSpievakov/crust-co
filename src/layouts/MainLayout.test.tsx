import '@testing-library/jest-dom';

import MainLayout from './MainLayout';
import rendererWithProviders from '../tests/helpers/rendererWithProviders';

describe('MainLayout Tests', () => {
  test('renders the MainLayout component', () => {
    const snapshot = rendererWithProviders(<MainLayout />);
    expect(snapshot).toMatchSnapshot();
  });
});
