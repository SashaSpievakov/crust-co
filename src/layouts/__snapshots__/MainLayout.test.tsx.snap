// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Tests renders the MainLayout component 1`] = `
Object {
  "0": <header
    className="sc-ftTHYK cCOKiD"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <section
        className="sc-gKPRtg bEDqgb"
      >
        <img
          alt="Pizza logo"
          height="54.5"
          src="main-logo.svg"
          width="38"
        />
        <div>
          <h1
            className="sc-iBYQkv kWKvOK"
          >
            Pizza Place
          </h1>
          <p>
            the most delicious pizzas in town
          </p>
        </div>
      </section>
    </a>
    <section
      className="sc-pyfCe kLYOGx"
    >
      <svg
        className="sc-jrcTuL eLShYk"
        data-testid="themeIcon"
        data-theme="light"
        fill="currentColor"
        height="1em"
        mode=""
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"
        />
      </svg>
      <a
        className="sc-ipEyDJ sc-eDWCr eVzfYL dGLzIP"
        data-testid="cartLink"
        href="/cart"
        onClick={[Function]}
      >
        <span>
          0
           $
        </span>
        <div
          className="sc-kDvujY ivIxim"
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .49.598l-1 5a.5.5 0 0 1-.465.401l-9.397.472L4.415 11H13a.5.5 0 0 1 0 1H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5zM3.102 4l.84 4.479 9.144-.459L13.89 4H3.102zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4zm-7 1a1 1 0 1 1 0 2 1 1 0 0 1 0-2zm7 0a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
          />
        </svg>
        <span>
          0
        </span>
      </a>
    </section>
  </header>,
  "1": <main
    className="sc-bcXHqe hvEOqJ"
  />,
  "2": <footer
    className="sc-gswNZR gHXegX"
  >
    <ul
      className="sc-dkrFOg cexZQQ"
    >
      <li>
        <a
          className="sc-hLBbgP ipdfCt"
          href="mailto:spievakov@gamil.com"
        >
          <svg
            className="sc-eDvSVe dQuWRI"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <span>
            spievakov@gmail.com
          </span>
        </a>
      </li>
      <li>
        <a
          className="sc-hLBbgP ipdfCt"
          href="tel:+16474739502"
        >
          <svg
            className="sc-eDvSVe dQuWRI"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
            />
          </svg>
          <span>
            +1 (647) 473-9502 
          </span>
        </a>
      </li>
      <li>
        <a
          className="sc-hLBbgP ipdfCt"
          href="https://www.google.com/maps/place/88+Alice+St,+Guelph,+ON+N1E+2Z8/@43.5471712,-80.2392008,17z/data=!3m1!4b1!4m5!3m4!1s0x882b9ae0d0802679:0x74ca5ec44983415f!8m2!3d43.5471673!4d-80.2370121"
          target="_blank"
        >
          <svg
            className="sc-eDvSVe dQuWRI"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
            />
          </svg>
          <span>
            88 Alice st
          </span>
        </a>
      </li>
    </ul>
    <p
      className="sc-jSUZER cUjmio"
    >
      Â© 2022 Oleksandr Spievakov
    </p>
  </footer>,
}
`;
